### PR TITLE
Supporting generic types through flytectl

### PR DESCRIPTION
**Pull request by pmahindrakar-oss**
Signed-off-by: Prafulla Mahindrakar <prafulla.mahindrakar@gmail.com>

# TL;DR

## Create task execution file with generic data structure 

`flytectl get task -d development -p flytectldemo  core.type_system.custom_objects.add --execFile adddatanum.yaml`

Output file created

```
iamRoleARN: ""
inputs:
  x: {}
  "y": {}
kubeServiceAcct: ""
targetDomain: ""
targetProject: ""
task: core.type_system.custom_objects.add
version: v2
```
## Modified task excution spec file with generic data 

```
iamRoleARN: "<iamrole>"
inputs:
  "x":
    "x": 2
    "y": ydatafory
    "z":
      1 : "foo"
      2 : "bar"
  "y":
    "x": 3
    "y": ydataforx
    "z":
      3 : "buzz"
      4 : "lightyear"
kubeServiceAcct: ""
targetDomain: ""
targetProject: ""
task: core.type_system.custom_objects.add
version: v3
```

## Creating execution with the modified file

```
flytectl create execution --execFile adddatanum.yaml  -p flytectldemo -d development
INFO[0000] Using config file: [/Users/praful/.flyte/config.yaml] 
INFO[0000] Config section [root] updated. No update handler registered.  src="viper.go:349"
INFO[0000] Config section [logger] updated. Firing updated event.  src="viper.go:351"

execution identifier project:"flytectldemo" domain:"development" name:"f07cac7db46364bbf831""
```

## Output for execution
```
flytectl get execution -p flytectldemo -d development f07cac7db46364bbf831        
INFO[0000] Using config file: [/Users/praful/.flyte/config.yaml] 
INFO[0000] Config section [storage] updated. No update handler registered.  src="viper.go:349"
INFO[0000] Config section [adminutils] updated. No update handler registered.  src="viper.go:349"
INFO[0000] Config section [root] updated. No update handler registered.  src="viper.go:349"
INFO[0000] Config section [logger] updated. Firing updated event.  src="viper.go:351"
 ---------------------- ------------------------------------- ------ ----------- ----------------------------- -------------- 
| NAME                 | LAUNCH PLAN NAME                    | TYPE | PHASE     | STARTED                     | ELAPSED TIME |
 ---------------------- ------------------------------------- ------ ----------- ----------------------------- -------------- 
| f07cac7db46364bbf831 | core.type_system.custom_objects.add | TASK | SUCCEEDED | 2021-06-01T10:56:40.137754Z | 12.748750s   |
 ---------------------- ------------------------------------- ------ ----------- ----------------------------- -------------- 
```

## flyteconsole 

Screenshot 2021-06-01 at 4 29 36 PM

## Type
 - [ ] Bug Fix
 - [X] Feature
 - [ ] Plugin

## Are all requirements met?

 - [X] Code completed
 - [ ] Smoke tested
 - [X] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/1055

## Follow-up issue
_NA_

### CODE REVIEW

File: cmd/create/execution.go [hunk 1]
```diff
@@ -78,6 +78,22 @@ Also an execution can be relaunched by passing in current execution id.
 
  flytectl create execution --relaunch ffb31066a0f8b4d52b77 -p flytectldemo -d development
 
+Generic data types are also supported for execution in similar way.Following is sample of how the inputs need to be escaped while creating the execution.
+Notice '>' used for escaping the json input which is passed as generic data for execution
+::
+
+  iamRoleARN: "arn:aws:iam::123456789:role/dummy"
+  inputs:
+    "x" : >
```
Comment (kumare3) [inline cmd/create/execution.go:87]:
```suggestion
    "x" : |
```

File: go.mod [hunk 2]
```diff
@@ -28,3 +28,5 @@ require (
 	gopkg.in/yaml.v2 v2.4.0
 	sigs.k8s.io/yaml v1.2.0
 )
+
+replace github.com/flyteorg/flyteidl => github.com/flyteorg/flyteidl v0.18.51-0.20210529142634-db6e9d4c701a
```
Comment (kumare3) [inline go.mod:32]: we should first check in flyteild

**Review by kumare3 — COMMENTED** [review]
lgtm, but lets get flyteidl checked in first

File: cmd/create/execution.go (hunk 1 above)
Comment (kumare3) [reply cmd/create/execution.go:87]: on second thought, i think this should rather be

```yaml
  x: 
    x: 2
    y: ydataforx
    z: {}
```

Comment (pmahindrakar-oss) [reply cmd/create/execution.go]: Done

File: go.mod (hunk 2 above)
Comment (pmahindrakar-oss) [reply go.mod]: Sure . Once this is approved and checkedin https://github.com/flyteorg/flyteidl/pull/176 i will update this file

**Comment by kumare3** [comment]
@pmahindrakar-oss lets update flyteidl and then this good to go?

**Comment by pmahindrakar-oss** [comment]
@kumare3 Done. Please have a look and merge if all looks ok.

**Review by kumare3 — APPROVED** [review]
